homeassistant/components/simplisafe/binary_sensor.py

```diff
@@ -0,0 +1,145 @@
+"""Support for SimpliSafe binary sensors."""
+from simplipy.entity import EntityTypes
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_MOISTURE,
+    DEVICE_CLASS_SMOKE,
+    BinarySensorEntity,
+)
+from homeassistant.core import callback
+
+from . import SimpliSafeEntity
+from .const import DATA_CLIENT, DOMAIN
+
+SUPPORTED_SENSOR_TYPES = [
+    EntityTypes.entry,
+    EntityTypes.carbon_monoxide,
+    EntityTypes.smoke,
+    EntityTypes.leak,
+]
+
+HA_SENSOR_TYPES = {
+    EntityTypes.entry: DEVICE_CLASS_DOOR,
+    EntityTypes.carbon_monoxide: DEVICE_CLASS_GAS,
+    EntityTypes.smoke: DEVICE_CLASS_SMOKE,
+    EntityTypes.leak: DEVICE_CLASS_MOISTURE,
+}
+
+SENSOR_MODELS = {
+    EntityTypes.entry: "Entry Sensor",
+    EntityTypes.carbon_monoxide: "Carbon Monoxide Detector",
+    EntityTypes.smoke: "Smoke Detector",
+    EntityTypes.leak: "Water Sensor",
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up SimpliSafe binary sensors based on a config entry."""
+    simplisafe = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
+
+    # Add sensor
+    async_add_entities(
+        [
+            SimpliSafeBinarySensor(simplisafe, system, sensor)
+            for system in simplisafe.systems.values()
+            for sensor in system.sensors.values()
+            if sensor.type in SUPPORTED_SENSOR_TYPES
+        ]
+    )
+
+    # Add low battery status entity for every sensor
+    async_add_entities(
+        [
+            SimpliSafeSensorBattery(simplisafe, system, sensor)
+            for system in simplisafe.systems.values()
+            for sensor in system.sensors.values()
+            if sensor.type in SUPPORTED_SENSOR_TYPES
+        ]
+    )
+
+
+class SimpliSafeBinarySensor(SimpliSafeEntity, BinarySensorEntity):
+    """Define a SimpliSafe binary sensor entity."""
+
+    def __init__(self, simplisafe, system, sensor):
+        """Initialize."""
+        super().__init__(simplisafe, system, sensor.name, serial=sensor.serial)
+        self._system = system
+        self._sensor = sensor
+        self._is_on = False
+
+    @property
+    def device_class(self):
+        """Return type of sensor."""
+        return HA_SENSOR_TYPES[self._sensor.type]
+
+    @property
+    def unique_id(self):
+        """Return unique ID of sensor."""
+        return f"{self._sensor.serial}-sensor"
```

Comment:
unique id is unique per platform, so you could skip it here and keep it only for the battery sensor